realscale.py

```diff
@@ -37,10 +37,43 @@ def __init__(self):
         self.OptionParser.add_option("--unit", action="store", type="string", dest="unit", default="cm",
                                      help="Real-world unit")
 
+    def calc_scale_center(self, p1x, p1y, p2x, p2y):
+        """ Use straight line as scaling center.
+            - determine which point is center on basis of quadrant the line is in.
+            - approx this by using center of line
+            0,0 corresponds to UL corner of page
+        """
+        scale_center = [0,0] # resulting scaling point
+        # calc page center
+        pagecenter_x = self.unittouu(self.document.getroot().get('width'))/2
+        pagecenter_y = self.unittouu(self.document.getroot().get('height'))/2
+        # calc minmax of straightline ref points
+        minx = min(p1x,p2x)
+        maxx = max(p1x, p2x)
+        miny = min(p1y,p2y)
+        maxy = max(p1y, p2y)
+        # simplifiy calc by using center of line to determine quadrant
```

Comment:
What was the reason for taking the straight line as a reference point (it's somewhat arbitrary, as it's just a part where the user knows the real length, but it could be anywhere inside the drawing to scale), instead of the corners or center of the object that is supposed to be scaled? 

(I would probably just have tried to find out how to scale around the center of the 'main object', or I might have chosen the left upper corner as a fixed point, so I'm interested why you chose this - it works great :)).